package_nso_to_oc/xe/xe_qos.py

```diff
@@ -0,0 +1,1432 @@
+                        "openconfig-qos:name": set_scheduler_policy_input(
+                            config_leftover, openconfig_scheduler, interface, intf, intf_to_sched_list, index)
+                    }
+                }
+            },
+            "openconfig-qos:output": {
+                "openconfig-qos:scheduler-policy": {
+                    "openconfig-qos:config": {
+                        "openconfig-qos:name": set_scheduler_policy_output(
+                            config_leftover, openconfig_scheduler, interface, intf, intf_to_sched_list, index)
+                    }
+                }
+            }
+        })
+
+
+def set_scheduler_policy_output(config_leftover, openconfig_scheduler, interface, intf, intf_to_sched_list, index):
+
+    scheduler_out = None
+    if "service-policy" in intf:
+        for int_index in intf_to_sched_list:
+            if "output" in intf["service-policy"] and intf["service-policy"]["output"] in int_index:
+                scheduler_out = int_index[intf['service-policy']['output']]
+                config_leftover["tailf-ned-cisco-ios:interface"][interface][index]["service-policy"]["output"] = None
+    
+    return scheduler_out
+
+
+def set_scheduler_policy_input(config_leftover, openconfig_scheduler, interface, intf, intf_to_sched_list, index):
+
+    scheduler_in = None
+    if "service-policy" in intf:
+        for intf_to_sched in intf_to_sched_list:
+            if "input" in intf["service-policy"] and intf["service-policy"]["input"] in intf_to_sched:
+                scheduler_in = intf_to_sched[intf['service-policy']['input']]
+                config_leftover["tailf-ned-cisco-ios:interface"][interface][index]["service-policy"]["input"] = None
+    
+    return scheduler_in
+
+
+def set_qos_class_map_ip(config_leftover, class_map_index, class_map, openconfig_class_map, policy_map_list):
+
+    if len(class_map["match"]["ip"]["dscp"]) == 1:
+        openconfig_class_map.append({
+            "openconfig-qos:name": class_map["name"],
+            "openconfig-qos:config": {"openconfig-qos:name:": class_map["name"],
+                                    "openconfig-qos:type:": "IPV4"},
+            "openconfig-qos:terms": {"openconfig-qos:term": [{"openconfig-qos:id": "term-" + f'{class_map_index + 1}',
+                                                            "openconfig-qos:config": {
+                                                                "openconfig-qos:id": "term-" + f'{class_map_index + 1}'
+                                                            },
+                                                            "openconfig-qos:conditions": {
+                                                                "openconfig-qos:ipv4": {
+                                                                    "openconfig-qos:config": {
+                                                                        "openconfig-qos:dscp": modify_dscp(class_map["match"]["ip"]["dscp"]),
```

Comment:
function modify_dscp returns a list. "openconfig-qos:dscp" should be a string

Reply:
Done